programs/clearing_house/src/math/liquidation.rs

```diff
@@ -0,0 +1,252 @@
+        )
+        .ok_or_else(math_error!())?
+        .checked_div(denominator_scale)
+        .ok_or_else(math_error!())
+}
+
+pub fn calculate_asset_transfer_for_liability_transfer(
+    asset_amount: u128,
+    asset_liquidation_multiplier: u128,
+    asset_decimals: u8,
+    asset_price: i128,
+    liability_amount: u128,
+    liability_liquidation_multiplier: u128,
+    liability_decimals: u8,
+    liability_price: i128,
+) -> ClearingHouseResult<u128> {
+    let (numerator_scale, denominator_scale) = if asset_decimals > liability_decimals {
+        (10_u128.pow((asset_decimals - liability_decimals) as u32), 1)
+    } else {
+        (1, 10_u128.pow((liability_decimals - asset_decimals) as u32))
+    };
+
+    let mut asset_transfer = liability_amount
+        .checked_mul(numerator_scale)
+        .ok_or_else(math_error!())?
+        .checked_mul(liability_price.unsigned_abs())
+        .ok_or_else(math_error!())?
+        .checked_mul(asset_liquidation_multiplier)
+        .ok_or_else(math_error!())?
+        .checked_div(
+            asset_price
+                .unsigned_abs()
+                .checked_mul(liability_liquidation_multiplier)
+                .ok_or_else(math_error!())?,
+        )
+        .ok_or_else(math_error!())?
+        .checked_div(denominator_scale)
+        .ok_or_else(math_error!())?;
+
+    // Need to check if asset_transfer should be rounded to asset amount
+    let (asset_value_numerator_scale, asset_value_denominator_scale) = if asset_decimals > 6 {
+        (10_u128.pow((asset_decimals - 6) as u32), 1)
+    } else {
+        (1, 10_u128.pow((asset_decimals - 6) as u32))
+    };
+
+    let asset_delta = if asset_transfer > asset_amount {
+        asset_transfer - asset_amount
+    } else {
+        asset_amount - asset_transfer
+    };
+
+    let asset_value_delta = asset_delta
+        .checked_mul(asset_price.unsigned_abs())
+        .ok_or_else(math_error!())?
+        .checked_div(MARK_PRICE_PRECISION)
+        .ok_or_else(math_error!())?
+        .checked_mul(asset_value_numerator_scale)
+        .ok_or_else(math_error!())?
+        .checked_div(asset_value_denominator_scale)
+        .ok_or_else(math_error!())?;
+
+    if asset_value_delta < QUOTE_PRECISION {
+        asset_transfer = asset_amount;
+    }
+
+    Ok(asset_transfer)
+}
+
+pub fn is_user_being_liquidated(
+    user: &User,
+    market_map: &MarketMap,
+    bank_map: &BankMap,
+    oracle_map: &mut OracleMap,
+    liquidation_margin_buffer_ratio: u8,
+) -> ClearingHouseResult<bool> {
+    let (margin_requirement, total_collateral) = calculate_margin_requirement_and_total_collateral(
+        user,
+        market_map,
+        MarginRequirementType::Maintenance,
+        bank_map,
+        oracle_map,
+    )?;
+
+    let margin_requirement_plus_buffer =
+        get_margin_requirement_plus_buffer(margin_requirement, liquidation_margin_buffer_ratio)?;
+
+    Ok(total_collateral >= cast(margin_requirement_plus_buffer)?)
```

Comment:
is the sign here backwards? shouldnt it be `<`

Reply:
yup good catch!